tools/flight_recorder/components/types.py

```diff
@@ -498,9 +498,21 @@ def match(self, other: "Op") -> MatchInfo:
                     f"Expected state: '{self.state}' does not match found state: '{other.state}'",
                 )
             if (
-                set(self.input_dtypes) != set(self.output_dtypes)
-                or set(self.input_dtypes) != set(other.input_dtypes)
-                or set(self.input_dtypes) != set(other.output_dtypes)
+                (
+                    set(self.input_dtypes) != set(self.output_dtypes)
+                    and self.input_sizes[0]
```

Comment:
should we be using the product of the size? i.e. (1, 0, 5) is size 0

Reply:
oh probably not because in the distributed_c10d.py, we just directly pass in an empty list in the list. input_sizes is like a list of list (you can see the example in the unit test). So the case you mention here won't be the case for gather/scatter.

https://github.com/pytorch/pytorch/blob/main/torch/distributed/distributed_c10d.py#L4151